frontend/packages/ux-editor/src/components/config/componentSpecificContent/AttachmentList/AttachmentListComponent.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import type { IGenericEditComponent } from '../../componentConfig';
+import { useAppMetadataQuery } from 'app-development/hooks/queries';
+import { useStudioUrlParams } from 'app-shared/hooks/useStudioUrlParams';
+import { useLayoutSetsQuery } from '../../../../hooks/queries/useLayoutSetsQuery';
+import { useAppContext } from '../../../../hooks/useAppContext';
+import type { ComponentType } from 'app-shared/types/ComponentType';
+import { useTranslation } from 'react-i18next';
+import {
+  reservedDataTypes,
+  convertInternalToExternalFormat,
+  convertExternalToInternalFormat,
+  selectionIsValid,
+} from './AttachmentListUtils';
+import type { AttachmentsFormat, InternalDataTypesFormat } from './AttachmentListUtils';
+import { AttachmentListInternalFormat } from './AttachmentListInternalFormat';
+import { StudioSpinner } from '@studio/components';
+import type { ApplicationMetadata } from 'app-shared/types/ApplicationMetadata';
+import type { LayoutSets } from 'app-shared/types/api/LayoutSetsResponse';
+
+export const AttachmentListComponent = ({
+  component,
+  handleComponentChange,
+}: IGenericEditComponent<ComponentType.AttachmentList>) => {
+  const { dataTypeIds = [] } = component || {};
+
+  const { t } = useTranslation();
+  const { org, app } = useStudioUrlParams();
+  const { data: layoutSets } = useLayoutSetsQuery(org, app);
+  const { data: appMetadata, isPending: appMetadataPending } = useAppMetadataQuery(org, app);
+  const { selectedLayoutSet } = useAppContext();
+
+  if (appMetadataPending)
+    return <StudioSpinner spinnerTitle={t('ux_editor.component_properties.loading')} />;
+
+  const availableAttachments: AttachmentsFormat = getAvailableAttachments(
+    layoutSets,
+    selectedLayoutSet,
+    appMetadata.dataTypes,
+  );
+
+  const onChange = (selectedDataTypes: InternalDataTypesFormat) => {
+    if (!selectionIsValid(selectedDataTypes)) return;
+
+    const resultingSelection = convertInternalToExternalFormat({
+      availableAttachments,
+      dataTypeIds: selectedDataTypes,
+    });
+
+    handleComponentChange({
+      ...component,
+      dataTypeIds: resultingSelection,
+    });
+  };
+
+  return (
+    <AttachmentListInternalFormat
+      onChange={onChange}
+      availableAttachments={availableAttachments}
+      internalDataFormat={convertExternalToInternalFormat({ availableAttachments, dataTypeIds })}
+    />
```

Comment:
Vedrørende forrige kommentar; du trenger ikke å kjøre validering her. Prøv gjerne å abstrahere ut ting her også, f.eks. slik:
```
  const handleChange = (internalDataFormat: InternalDataTypesFormat) => {
    const externalDataFormat = convertInternalToExternalFormat(availableAttachments, internalDataFormat);
    const updatedComponent = setDataTypes(component, externalDataFormat);
    handleComponentChange(updatedComponent);
  };

  const { dataTypeIds = [] } = component || {};
  const internalDataFormat = convertExternalToInternalFormat(availableAttachments, dataTypeIds);

  return (
    <AttachmentListInternalFormat
      onChange={handleChange}
      availableAttachments={availableAttachments}
      internalDataFormat={internalDataFormat}
    />
  );
```
Her kan du også se at jeg har flyttet `dataTypesIds` lengre ned. Det er for å få den til å stå nærmere der den brukes.